pontoon/sync/vcs/models.py

```diff
@@ -37,6 +40,54 @@
 log = logging.getLogger(__name__)
 
 
+class DownloadTOMLParser(TOMLParser):
+    """
+    This wrapper is a workaround for the lack of the shared and persistent filesystem
+    on Heroku workers.
+    Related: https://bugzilla.mozilla.org/show_bug.cgi?id=1530988
+    """
+
+    def __init__(self, checkout_path, permalink_prefix):
+        self.checkout_path = checkout_path
+        self.permalink_prefix = permalink_prefix
+
+    def get_download_path(self, path):
+        """Return the directory in which the config file should be stored."""
+        return os.path.join(self.checkout_path, path)
+
+    def get_remote_path(self, path):
+        """Construct the link to the remote resource based on the local path."""
+        config_path = path.replace(self.checkout_path, "")
+
+        if "{locale_code}" in self.permalink_prefix:
+            return self.permalink_prefix.format(locale_code=config_path)
+        else:
+            return urljoin(self.permalink_prefix, config_path)
+
+    def get_project_config(self, path):
+        """Download the project config file and return its local path."""
+        download_path = self.get_download_path(path)
+
+        with open(download_path, "wb") as f:
```

Comment:
We need to make sure we create the file if it doesn't exist yet. Most of the time it will NOT exist on the `web` worker on prod and will raise an error similar to this one:
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.8/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/local/lib/python3.8/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.8/contextlib.py", line 75, in inner
    return func(*args, **kwds)
  File "/app/pontoon/base/views.py", line 642, in download
    content, filename = utils.get_download_content(slug, code, part)
  File "/app/pontoon/base/utils.py", line 238, in get_download_content
    source_path = _download_file(
  File "/app/pontoon/base/utils.py", line 144, in _download_file
    absolute_l10n_path = vcs_project.configuration.l10n_path(
  File "/app/pontoon/sync/vcs/models.py", line 653, in l10n_path
    project_files = self.get_or_set_project_files(locale.code)
  File "/app/pontoon/sync/vcs/models.py", line 641, in get_or_set_project_files
    and locale_code not in self.parsed_configuration.all_locales
  File "/usr/local/lib/python3.8/site-packages/django/utils/functional.py", line 80, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
  File "/app/pontoon/sync/vcs/models.py", line 590, in parsed_configuration
    return DownloadTOMLParser(
  File "/app/pontoon/sync/vcs/models.py", line 85, in parse
    self.get_project_config(path), env, ignore_missing_includes
  File "/app/pontoon/sync/vcs/models.py", line 74, in get_project_config
    with open(download_path, "wb") as f:
FileNotFoundError: [Errno 2] No such file or directory: '/app/media/projects/project-config/mathjazz/moz-l10n-config-test.git/l10n.toml'
```

Reply:
Ideally, we should only download the config file if it's not available yet.